modal/cli/network_file_system.py

```diff
@@ -188,7 +193,41 @@ async def get(
     ensure_env(env)
     destination = Path(local_destination)
     volume = await _volume_from_name(volume_name)
-    await _volume_download(volume, remote_path, destination, force)
+    if "*" in remote_path:
+        await _volume_download(volume, remote_path, destination, force)
+        return
+
+    if destination != PIPE_PATH:
+        if destination.is_dir():
+            destination = destination / remote_path.rsplit("/")[-1]
+
+        if destination.exists() and not force:
+            raise UsageError(f"'{destination}' already exists")
+
+        if not destination.parent.exists():
+            raise UsageError(f"Local directory '{destination.parent}' does not exist")
+
+    @contextmanager
+    def _destination_stream():
+        if destination == PIPE_PATH:
+            yield sys.stdout.buffer
+        else:
+            with NamedTemporaryFile(delete=False) as fp:
+                yield fp
+            shutil.move(fp.name, destination)
+
```

Comment:
Can we fix `_volume_download` instead of introducing deviant behavior based on the path containing an asterisk?